### PR TITLE
T4: convert IsEmpty/IsNotEmpty Facts to Theories + add capacity tests (closes #117)

### LINKED ISSUE #117: [Maintenance] testing: Audit test quality
## Scope

Audit the existing test suite for quality issues that pass CI but reduce real signal: tests that do not assert, weak assertions where stronger ones exist (Assert.True(x != null) -> Assert.NotNull), Assert.Equal(null, ...) instead of Assert.Null, test naming pattern violations (MethodUnderTest_when_condition_expected_result), shared mutable state across test runs, tests of implementation details rather than behavior, hardcoded magic numbers (e.g. 9999999L -> TimeSpan.TicksPerSecond - 1), [Theory] inline data that should be MemberData, missing boundary-case tests for paths the source explicitly handles (MinValue/MaxValue clamps, null-input short-circuits). Run as Phase 4.5 of the code-review skill.

## Acceptance criteria

Each finding categorized as must-fix / should-fix / nit. Must-fix and should-fix items either filed as separate sub-issues or addressed inline. nit items reviewed and either addressed, deferred, or closed as intentional style preference.

## Notes

- Tier-1 foundation task — part of fleet initiative **T4: Audit test quality**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Summary

Closes #117 (audit test quality).

Converts the two compound-assertion `[Fact]` tests for `IsEmpty` /
`IsNotEmpty` into `[Theory]` + `MemberData` with 8 collection-type ×
empty-state cases each. Each case asserts exactly one thing.

### IsEmpty / IsNotEmpty
Before: 4 `[Fact]`s with multiple `Assert.*` calls bundled per
collection type.
After: 2 `[Theory]`s with 8 rows each covering:
- `List<int>` empty / non-empty
- `HashSet<int>` empty / non-empty
- `LinkedList<int>` empty / non-empty (via new `LinkedListWith` helper)
- `Collection<int>` empty / non-empty

### New pre-allocation tests
Two new `[Fact]`s pin the `AddRange` capacity fast path:
1. `AddRange_when_source_is_List_and_items_is_ICollection_pre_allocates_capacity`
   — proves `Capacity` lands at exactly the final `Count`, not at the
   next power of two that amortised doubling would produce.
2. `AddRange_when_source_List_already_has_sufficient_capacity_does_not_shrink_it`
   — pins the `Math.Max(Capacity, needed)` guard.

## Local verification

`dotnet test --framework net10.0`: **39 passed**, 0 failed (up from
~25 in the previous shape — the 16-row Theory replaces 4 Facts and
the 2 new pre-allocation Facts).

## Stacking

Top of the vNext stack. Bases on `feature/c5-quickwins` (PR #126).

### CODE REVIEW

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 1]
```diff
@@ -376,16 +392,89 @@ public void IsNotEmpty_when_source_has_items_returns_true()
...
+        // relying on List's amortised doubling.
+        var source = new List<int>(capacity: 1) { 0 };
+        var items = new List<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
 
-        // Act & Assert
-        Assert.False(empty.IsNotEmpty());
-        Assert.True(nonEmpty.IsNotEmpty());
+        // Act
+        source.AddRange(items);
+
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs]: Fixed in d1160e8 — receiver is now typed as ICollection<int> so the extension wins overload resolution. We keep a 'List<int> list' reference for the Capacity assert; inside the extension the runtime 'source is List<T>' check still succeeds so the pre-allocation branch fires.

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 2]
```diff
@@ -376,16 +392,89 @@ public void IsNotEmpty_when_source_has_items_returns_true()
...
+    [Fact]
+    public void AddRange_when_source_List_already_has_sufficient_capacity_does_not_shrink_it()
+    {
+        // Arrange — list with deliberately oversized capacity.
+        var source = new List<int>(capacity: 100) { 0 };
+        var items = new List<int> { 1, 2, 3 };
+
+        // Act
+        source.AddRange(items);
+
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs]: Fixed in d1160e8 — same fix as the test above (ICollection<int> receiver, List<int> reference held for the Capacity assert). The Math.Max(current, needed) guard is now actually exercised.
